### PR TITLE
Add an order parameter to /v2/content

### PR DESCRIPTION
**Pull request by tuzz**
This pull request adds an order parameter to allow publishing apps to order their results by a field other than `public_updated_at` which was previously hardcoded. We have used the same interface as documented [here](https://github.com/alphagov/rummager/blob/master/docs/unified-search-api.md) as suggested by @tijmenb.

This work was completed as part of this ticket:
https://trello.com/c/XJRTeF9f/130-spv2-orders-by-the-wrong-timestamp-on-the-search-index

We decided to limit the fields that you may order by to those that had indexes and left a comment indicating that we should be careful to add an index to any additional fields with which we wish to order. Otherwise, performance would be affected when running the query.

### CODE REVIEW

**Comment by benilovj** [comment]
The order direction behaviour feels a bit magic and inflexible (what if you want to order descending on a custom field). Is there a good reason to disallow setting it through a parameter as well?

**Comment by tuzz** [comment]
Please could you elaborate? What do you mean by custom field? Are you referring to a field such as `publication_state` which is an alias of `states.name` ?

**Comment by benilovj** [comment]
Actually, ignore me - I misread the code. It's fine :+1:

File: spec/controllers/v2/content_items_controller_spec.rb [hunk 1]
```diff
@@ -177,6 +177,93 @@
...
+          results = JSON.parse(response.body)["results"]
+
+          expect(results).to eq([
+            { "base_path" => "/content.ar" },
+            { "base_path" => "/content.en" },
+          ])
+        end
+      end
+
+      context "when ordering by base_path ascending" do
```
Comment (benilovj) [inline spec/controllers/v2/content_items_controller_spec.rb]: Should this say 'descending'?

File: spec/controllers/v2/content_items_controller_spec.rb [hunk 2]
```diff
@@ -177,6 +177,93 @@
...
+          results = JSON.parse(response.body)["results"]
+
+          expect(results).to eq([
+            { "updated_at" => "2016-01-01 00:00:00" },
+            { "updated_at" => "2016-02-02 00:00:00" },
+          ])
+        end
+      end
+
+      context "when ordering by updated_at ascending" do
```
Comment (benilovj) [inline spec/controllers/v2/content_items_controller_spec.rb]: this should say "when ordering by updated_at descending" ?

**Pull request merged**